xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -2269,8 +2276,16 @@ private Any getRawResource() {
       restartTimer();
     }
 
-    // TODO(zdapeng): add resourceName arg and support xdstp:// resources
-    private ServerInfo getServerInfo() {
+    private ServerInfo getServerInfo(String resource) {
+      if (resource.startsWith(XDSTP_SCHEME)) {
+        URI uri = URI.create(resource);
+        String authority = uri.getAuthority();
+        if (authority == null) {
+          authority = "";
+        }
+        AuthorityInfo authorityInfo = bootstrapInfo.authorities().get(authority);
+        return authorityInfo.xdsServers().get(0);
```

Comment:
but this can be empty, isn't it?

Reply:
Good point. Clarified in the javadoc of the API it must be non-empty.

Reply:
oh, gRFC says it can be empty in an example:

> The xds_servers list is not specified in the entry for the authority, so we default to the top-level xds_servers list, which is the same one that we would have used prior to this design.

Reply:
Yeah, that's the json field. For the parsed data class, I define it to default to top-level xds_servers and never empty.